MatrixSDK/Space/MXSpaceStore.swift

```diff
@@ -0,0 +1,124 @@
+// 
+// Copyright 2021 The Matrix.org Foundation C.I.C
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Foundation
+
+/// `MXSpaceStore` is used to store the spaces related data into a permanent store
+class MXSpaceStore {
+    
+    // MARK: - Constants
+    
+    private enum Constants {
+        static let fileStoreFolder = "MXSpaceStore"
+        static let fileStoreGraphFile = "graph"
+        static let backupFileExtension = "backup"
+    }
+    
+    // MARK: - Properties
+
+    private let userId: String
+    private var storeUrl: URL?
+
+    // MARK - Setup
+    
+    init(session: MXSession) {
```

Comment:
Please use userId/deviceId instead of a dependency on MXSession.
DeviceId is important because of integration tests that can run 2 devices for the same user in parallel.